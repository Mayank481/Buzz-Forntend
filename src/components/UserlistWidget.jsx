import React from 'react';
import { Link } from 'react-router-dom';

function UserlistWidget({ title, friendList, ifEmpty }) {
  return (
    <div className="border p-2 mb-3 scroll bg-white shadow-lg p-3 bg-body rounded border-0 ">
      <div className="d-flex justify-content-between">
        <h6>{title}</h6>
       
      </div>
      {friendList.length === 0 && <div className="text-center">{ifEmpty}</div>}

      {friendList.map((friend,index) => (
        <Link
        key={index}
          to={'/profile/' + friend._id}

          className="d-flex mt-2 text-decoration-none "
        >
          <div>
            <img
              src={friend.picture_url || require('../images/blank-profile.png')}
              className="card-img-top small-round-pic me-2  round-img"
              alt="..."
            />
          </div>
          <div data-testid="friendName" className="d-flex align-items-center text-dark">
            {'firstname' in friend
              ? friend.firstname + ' ' + friend.lastname
              : 'Unknown User'}
          </div>
        </Link>
      ))}
      {/* <li>hii</li>
      <li>hii</li>
      <li>hii</li>
      <li>hii</li>
      <li>hii</li>
      <li>hii</li>
      <li>hii</li>
      <li>hii</li>
      <li>hii</li>
      <li>hii</li>
      <li>hii</li>
      <li>hii</li> */}
    </div>
  )
}

export default UserlistWidget;
