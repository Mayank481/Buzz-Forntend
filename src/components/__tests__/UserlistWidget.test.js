import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import UserlistWidget from "../UserlistWidget";
import renderer from 'react-test-renderer';



test('should render UserlistWidget Panel', () => {
  const user = { friends: { myFriends: [], mySentRequests: [{}] }, suggestFriend: [{ firstname: "kumar", lastname: "kumar" }] };

  const toggleRefresh = () => setRefresh((p) => !p);
  render(<BrowserRouter> <UserlistWidget
    title={'Friends Sugesstions'}
    friendList={user.suggestFriend}
    ifEmpty="No Suggestions found"
  /> </BrowserRouter>)
  const text = screen.queryByTestId('friendName').innerHTML;
  expect(text).toBe("kumar kumar");

  const component = renderer.create(<BrowserRouter> <UserlistWidget
    title={'Friends Sugesstions'}
    friendList={user.suggestFriend}
    ifEmpty="No Suggestions found"
  /> </BrowserRouter>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();

})