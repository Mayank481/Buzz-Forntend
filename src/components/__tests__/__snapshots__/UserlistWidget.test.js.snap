// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserlistWidget Panel 1`] = `
Array [
  " ",
  <div
    className="border p-2 mb-3 scroll bg-white shadow-lg p-3 bg-body rounded border-0 "
  >
    <div
      className="d-flex justify-content-between"
    >
      <h6>
        Friends Sugesstions
      </h6>
    </div>
    <a
      className="d-flex mt-2 text-decoration-none "
      href="/profile/undefined"
      onClick={[Function]}
    >
      <div>
        <img
          alt="..."
          className="card-img-top small-round-pic me-2  round-img"
          src="blank-profile.png"
        />
      </div>
      <div
        className="d-flex align-items-center text-dark"
        data-testid="friendName"
      >
        kumar kumar
      </div>
    </a>
  </div>,
  " ",
]
`;
